File: example - copia/java/deductions.java
public class deductions {
    protected double healthRate;
    protected double pensionRate;
    protected double arlRate;

    public deductions(double healthRate, double pensionRate, double arlRate) {
        this.healthRate = healthRate;
        this.pensionRate = pensionRate;
        this.arlRate = arlRate;
    }

    public deductions() {
        this(0.12, 0.16, 0.052); // Default values
    }

    public double discounts(double salary) {
        return 0; // This method will be overridden in child classes
    }

    float getHealthRate() {
        throw new UnsupportedOperationException("Not supported yet.");
    }

    void setHealthRate(float healthRate) {
        throw new UnsupportedOperationException("Not supported yet.");
    }
}

File: example - copia/deductions.php
<?php

class deductions {
    protected $healthRate;
    protected $pensionRate;
    protected $arlRate;

    public function __construct(float $healthRate = 0.12, float $pensionRate = 0.16, float $arlRate = 0.052) {
        $this->healthRate = $healthRate;
        $this->pensionRate = $pensionRate;
        $this->arlRate = $arlRate;
    }

    public function discounts(float $salary): float {
        return 0; // Este método se sobrescribirá en las clases hijas
    }
}

?>